### ISSUE TITLE
[bridges] affine vs cartesian dependency on bridges

### ISSUE THREAD

**Issue by antoinevanmuylder** [antoinevanmuylder/agda#3]
As of now the `--bridges` branch aims to adapt the type theory described in section 2 of [C-H](https://arxiv.org/pdf/2005.11290.pdf) and to integrate it in Agda.
The context comprehension for bridge variables `Γ ctx , (x : BI) ↦ Γ, x:BI ctx` appearing in C-H does not correspond to the naive context comprehension of Agda, but rather to an affine context comprehension. Semantically comprehension for bridge variables in C-H is always interpreted as a tensor product (day convolution) `Γ ⊗ BI` instead of a cartesian product `Γ × BI`.
To emulate this affine context comprehension in our implementation we need the following:
1. In Agda files all the bridge vars should be systematically tagged with a tick annotation
   ```agda
   A : (@tick x : BI) → Set ℓ   --example1: all type lines should be ticked
   Gel : A0 A1 R (@tick r : BI) → Set ℓ --example2 This incidentally allows to state freshness constraints by typing.
                                         --freshness: "A0 A1 R are apart from r"
   ```
2. In the typechecker, whenever a bridge var appears as an `Arg`, its argInfo has a lock annotation (lock,tick are synonyms). This is probably more complex than this..

At the time of writing (2) is not satisfied yet.

**Comment by antoinevanmuylder** [comment]
@Saizan
- Do you think (2) would be an "easy" fix?
- Would it prevent us from considering the applications you were mentionning involving the cartesian functions `(x : BI) ↦ A x` instead of the affine functions `(@tick x : BI) ↦ A x`

**Comment by Saizan** [comment]
I don't think (2) is what we want.

As I see it the situation is that 
- affine context extension and affine exponential with bridge variables is working fine in isolation.
- Cartesian context extension and cartesian exponential with bridge variables is broken.

CH does not have the latter so the easy solution would be to make the typechecker reject it: we would not need to add any automatic annotation for this, we just have to throw a type error for `(x : BI) -> A x`.

If we want to keep `(x : BI) -> A x` around instead we have to fix the typing rule for affine tick application. 
In the case of the bridge interval I think it would be sound to have that for `t r` to typecheck when `t` is a bridge or an affine exponential we want `r` to only mention affine bridge variables.

**Comment by antoinevanmuylder** [comment]
@Saizan 

> Cartesian context extension and cartesian exponential with bridge variables is broken.

I think it is broken for tick variables in general?
`yes-diag` below should be unsound

```agda
{-# OPTIONS --cubical --guarded #-}
open import Agda.Builtin.Cubical.Path

module LaterPrimsBug where

primitive
  primLockUniv : Set₁

postulate
  Tick : primLockUniv

module tickTest {ℓ} {S : Set ℓ} where
 
  -- no-diag : ( (@tick α β : Tick) → S ) → (@tick x : Tick) → S
  -- no-diag p x = p x x  -- does not typecheck

  toAff : (Tick → S) → (@tick x : Tick) → S
  toAff f x = f x

  yes-diag : ( (@tick α β : Tick) → S ) → (@tick x : Tick) → S
  yes-diag p = toAff ( λ x → p x x ) --this one typechecks :(

  -- the above normalizes to λ p x → p x x   which is equal to no-diag

  toCart : ( (@tick x : Tick) → S ) → Tick → S
  toCart f x = f x

  cart-retract : ∀ (f : Tick → S) → toCart (toAff f) ≡ f
  cart-retract = λ f i → f

  aff-retract : ∀ (f : (@tick x : Tick) → S) → toAff (toCart f) ≡ f
  aff-retract = λ f i → f
```

**Comment by Saizan** [comment]
Yes, I was focused on the parametricity part of this, but indeed it's generally broken. 
With --guarded though it's simple: the exponential does not make sense at all.

**Comment by antoinevanmuylder** [comment]
ok. so it would make sense to throw an error at `(x : Tick) -> S` in guarded-cubical as well
